server/src/main/java/io/deephaven/server/arrow/FlightServiceGrpcImpl.java

```diff
@@ -109,6 +131,18 @@ public void onNext(final Flight.HandshakeRequest value) {
                 auth = login(BasicAuthMarshaller.AUTH_TYPE, protocolVersion, payload, handshakeResponseListener);
                 if (auth.isEmpty()) {
                     final WrappedAuthenticationRequest req = WrappedAuthenticationRequest.parseFrom(payload);
+                    // If the auth request is bearer, the v1 auth might be trying to renew an existing session
+                    if (req.getType().equals(Auth2Constants.BEARER_PREFIX.trim())) {
+                        try {
+                            UUID uuid = UuidCreator.fromString(req.getPayload().toString(StandardCharsets.US_ASCII));
+                            SessionState session = sessionService.getSessionForToken(uuid);
+                            respondWithAuthTokenBin(session);
```

Comment:
I'm not sure how much we care right now, and I'm assuming we have similar races elsewhere, but there's potential for `SessionService#getSessionForToken` to succeed and for `respondWithAuthTokenBin` to NPE if `SessionState#getExpiration` has expired.

Reply:
That's true - I made a quick change here that I think corrects this.